### PR TITLE
Notification body structure

### PR DESCRIPTION
**Pull request by raonismaneoto**
**Feature/Bug description:**
Improving the push_notification's creation.

**Solution:**
Create a service that provides the properties based on the notification's type.

**TODO/FIXME:** n/a

### CODE REVIEW

File: backend/fcm.py [hunk 1]
```diff
@@ -57,10 +57,14 @@ def send_push_notifications(title, body, tokens):
         tokens: The devices' tokens that will receive
         the notification.
     """
+    title = data['title']
```
Comment (JuliePessoa) [inline backend/fcm.py]: theres no need to validate if really contains these properties?

File: backend/handlers/institution_children_request_collection_handler.py [hunk 2]
```diff
@@ -72,4 +74,9 @@ def post(self, user, institution_urlsafe):
 
         request.send_invite(host, user.current_institution)
 
+        requested_inst = requested_inst_key.get()
```
Comment (JuliePessoa) [inline backend/handlers/institution_children_request_collection_handler.py]: extract to a method this block that send the push notification

File: backend/handlers/institution_parent_request_collection_handler.py [hunk 3]
```diff
@@ -125,4 +127,9 @@ def main_operations(request, requested_inst_key, child_institution, user, host):
 
         request = main_operations(request, requested_inst_key, child_institution, user, host)
 
+        requested_inst = requested_inst_key.get()
```
Comment (JuliePessoa) [inline backend/handlers/institution_parent_request_collection_handler.py]: extract to a method this block that send the push notification

File: backend/test/util_modules_tests/fcm_test.py [hunk 4]
```diff
@@ -25,19 +25,27 @@ def setUp(cls):
...
         get_token.assert_called()
         send_notification.assert_called()
     
     @patch('fcm.send_push_notifications')
     @patch('fcm.get_multiple_user_tokens')
     def test_notify_multiple_users(self, get_tokens, send_notification):
-        title = 'test'
-        body = 'test'
-        fcm.notify_multiple_users(title, body, [self.f_user_key, self.s_user_key])
+        notification_props = {
```
Comment (JuliePessoa) [inline backend/test/util_modules_tests/fcm_test.py]: its the same object above. make it a class object

File: backend/worker.py [hunk 5]
```diff
@@ -423,10 +426,18 @@ def post(self):
...
         current_institution = self.request.get('current_institution')
         current_institution = ndb.Key(urlsafe=current_institution)
+        type_of_invite = self.request.get('type_of_invite')
+        user_keys = []
 
         for key in keys:
             invite = ndb.Key(urlsafe=key).get()
             invite.send_invite(host, current_institution)
+            user = User.get_active_user(invite.invitee)
+            user = user.key.urlsafe() if user else None
```
Comment (JuliePessoa) [inline backend/worker.py]: `userKey = user.key.urlsafe() if user else None`

File: backend/test/util_modules_tests/push_notification_service_test.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        cls.inst = mocks.create_institution()
+        cls.post = mocks.create_post(cls.f_user.key, cls.inst.key)
+    
+
+    def test_like_post_notification(self):
+        url = "/posts/%s" %self.post.key.urlsafe()
+
+        notification_props = get_notification_props('LIKE_POST', self.post)
+
+        self.assertEqual(notification_props['title'], 'Publicação curtida')
```
Comment (Luiz-FS) [inline backend/test/util_modules_tests/push_notification_service_test.py]: Add assertion message

File: backend/test/util_modules_tests/push_notification_service_test.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        self.assertEqual(
```
Comment (Luiz-FS) [inline backend/test/util_modules_tests/push_notification_service_test.py]: Same

File: backend/test/util_modules_tests/push_notification_service_test.py [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+        cls.policy = cls.datastore.PseudoRandomHRConsistencyPolicy(
+            probability=1)
+        cls.test.init_datastore_v3_stub(consistency_policy=cls.policy)
+        cls.test.init_memcache_stub()
+        cls.f_user = mocks.create_user()
+        cls.inst = mocks.create_institution()
+        cls.post = mocks.create_post(cls.f_user.key, cls.inst.key)
+    
+
+    def test_like_post_notification(self):
```
Comment (Luiz-FS) [inline backend/test/util_modules_tests/push_notification_service_test.py]: Add Docstring of test

File: backend/test/util_modules_tests/push_notification_service_test.py [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+        url = "/posts/%s" %self.post.key.urlsafe()
+
+        notification_props = get_notification_props('LIKE_POST', self.post)
+
+        self.assertEqual(notification_props['title'], 'Publicação curtida')
+        self.assertEqual(
+            notification_props['body'], 'Uma publicação de seu interesse foi curtida')
+        self.assertEqual(notification_props['click_action'], url)
+    
+    def test_like_post_notification_without_entity(self):
```
Comment (Luiz-FS) [inline backend/test/util_modules_tests/push_notification_service_test.py]: Same

File: backend/test/util_modules_tests/push_notification_service_test.py [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+        with self.assertRaises(EntityException) as ex:
+            notification_props = get_notification_props('LIKE_POST')
+        
+        error_message = ex.exception.message
+
+        self.assertEqual(
```
Comment (Luiz-FS) [inline backend/test/util_modules_tests/push_notification_service_test.py]: Add assertion message

File: backend/handlers/institution_parent_request_collection_handler.py (hunk 3 above)
Comment (Luiz-FS) [inline backend/handlers/institution_parent_request_collection_handler.py]: The sending of notification should be in a queue of tasks, not to cause a bottleneck in the client's request.

File: backend/test/invite_handlers_tests/invite_user_collection_handler_test.py [hunk 11]
```diff
@@ -86,15 +86,24 @@ def test_post_invite_user(self, enqueue_task, verify_token, create_sent_invites_
...
-            }
-        )
+        send_invite_params = {
+            'invites_keys': json.dumps([invite.key.urlsafe()]),
+            'host': response.request.host,
+            'current_institution': institution.key.urlsafe(),
+            'notifications_ids': ['some_notification_id']
+        }
+
+        send_push_not_params = {
```
Comment (JuliePessoa) [inline backend/test/invite_handlers_tests/invite_user_collection_handler_test.py]: not?

File: backend/test/worker_handlers_tests/send_push_notification_handler_test.py [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+from push_notification import SendPushNotificationHandler, get_notification_props, NotificationType
+import json
+from .. import mocks
+from mock import patch
+
+MAIN_URI = '/api/queue/send-push-notification'
+
+
+class SendPushNotificationHandlerTest(TestBaseHandler):
+    """Test Remove admin permission in institution hierarchy."""
```
Comment (JuliePessoa) [inline backend/test/worker_handlers_tests/send_push_notification_handler_test.py]: update doc

File: backend/worker.py [hunk 13]
```diff
@@ -3,7 +3,8 @@
 import webapp2
 import json
 from firebase import send_notification
-from fcm import notify_single_user
+from fcm import notify_single_user, notify_multiple_users
```
Comment (JuliePessoa) [inline backend/worker.py]: is this two lines of imports being used?

File: backend/push_notification/push_notification_service.py [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+
+    Args:
+        _type -- the notification's type. type: NotificationType
+        entity -- an optional parameter that can be
+        used to determine the click_action property 
+    """
+    notification = NotificationProperties(_type, entity)
+    return notification.get_props()
+
+class NotificationType(Enum):
```
Comment (Luiz-FS) [inline backend/push_notification/push_notification_service.py]: Is it necessary to extend enum?

Comment (raonismaneoto) [reply backend/push_notification/push_notification_service.py]: Yes, I need it

**Pull request merged**